sdcclient/_scanning.py

```diff
@@ -0,0 +1,535 @@
+        """
+        SUBSCRIPTION_TYPE: Type of subscription. Valid options:
+
+          - tag_update: Receive notification when new image is pushed
+
+          - policy_eval: Receive notification when image policy status changes
+
+          - vuln_update: Receive notification when vulnerabilities are added, removed or modified
+
+        SUBSCRIPTION_KEY: Fully qualified name of tag to subscribe to. Eg. docker.io/library/alpine:latest
+        """
+        return apiexternal.activate_subscription(self._anchore_config, subscription_type, subscription_key)
+
+    def deactivate_subscription(self, subscription_type, subscription_key):
+        """
+        SUBSCRIPTION_TYPE: Type of subscription. Valid options:
+
+          - tag_update: Receive notification when new image is pushed
+
+          - policy_eval: Receive notification when image policy status changes
+
+          - vuln_update: Receive notification when vulnerabilities are added, removed or modified
+
+        SUBSCRIPTION_KEY: Fully qualified name of tag to subscribe to. Eg. docker.io/library/alpine:latest
+        """
+        return apiexternal.deactivate_subscription(self._anchore_config, subscription_type, subscription_key)
+
+    def list_subscription(self):
+        '''**Description**
+            List all subscriptions
+
+        **Arguments**
+            - None
+
+        **Success Return Value**
+            A JSON object representing the list of subscriptions.
+        '''
+        return apiexternal.get_subscription(self._anchore_config)
+
+    def add_registry(self, registry, registry_user, registry_pass, insecure=False, registry_type="docker_v2", validate=True):
+        '''**Description**
+            Add image registry
+
+        **Arguments**
+            - registry: Full hostname/port of registry. Eg. myrepo.example.com:5000
+            - registry_user: Username
+            - registry_pass: Password
+            - insecure: Allow connection to registry without SSL cert checks (ex: if registry uses a self-signed SSL certificate)
+            - registry_type: Specify the registry type. 'docker_v2' and 'awsecr' are supported (default='docker_v2')
+            - validate: If set to 'False' will not attempt to validate registry/creds on registry add
+
+        **Success Return Value**
+            A JSON object representing the registry.
+        '''
+        registry_types = ['docker_v2', 'awsecr']
+
+        if registry_type and registry_type not in registry_types:
+            return [False, "input registry type not supported (supported registry_types: " + str(registry_types)]
+
+        # try to detect awsecr registry of form <accid>.dkr.ecr.<region>.amazonaws.com
+        if not registry_type:
+            if re.match("[0-9]+\\.dkr\\.ecr\\..*\\.amazonaws\\.com", registry):
+                registry_type = "awsecr"
+            else:
+                registry_type = "docker_v2"
+
+        # do some input string checking
+        if re.match(".*\\/.*", registry):
```

Comment:
We don't need a comment if we refactor this regex:
`if not self._registry_string_is_valid(registry):`

Moreover, we are using this code four times in the code. So this refactor is totally worthy.